lnwallet/mock.go

```diff
@@ -397,3 +399,45 @@ func (*mockChainIO) GetBlockHeader(
 
 	return nil, nil
 }
+
+type MockAuxLeafStore struct{}
```

Comment:
should use `mock.Mock` instead and test the error cases - it's also on my todo list to replace the mocks with `mock.Mock`, so maybe I will change it there.

Reply:
The goal with these mocks was to trigger all the `fn.MapOptionZ(auxComponentXYZ, ...)` since they wouldn't be `None` when a mock was given. To test that nothing breaks.

And to be honest, I'm not yet super familiar with `mock.Mock`, so I'd prefer not doing it in this PR. I'm happy to help convert old mocks to `mock.Mock` once this series of PRs is in though.